tools/crm_resource.c

```diff
@@ -1005,6 +1021,22 @@ initialize_scheduler_data(xmlNodePtr *cib_xml_copy)
     return pcmk_rc_ok;
 }
 
+static void
+list_options(void)
+{
+    if (pcmk__str_eq(options.opt_list, PCMK_VALUE_FENCING, pcmk__str_none)) {
+        exit_code = pcmk_rc2exitc(pcmk__list_fence_params(out, options.all));
+
+    } else {
+        // @TODO Improve usage messages to reduce duplication
+        exit_code = CRM_EX_USAGE;
+        g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_USAGE,
+                    "Invalid --list-options value '%s'. Allowed values: "
+                    PCMK_VALUE_FENCING,
+                    pcmk__s(options.opt_list, "(BUG: none)"));
```

Comment:
It's better to do validation in an option callback when practical, so invalid arguments are handled consistently (by glib). In this case we could save the filter rather than the text in options

Reply:
Yeah I think validation should take precedence over last-wins, since otherwise it could obscure an error (consider a script that incorrectly adds a value intended to be used as a default, then sometimes correctly adds a different value after it)